traffic_ops/testing/api/v4/cachegroups_parameters_test.go

```diff
@@ -79,6 +90,129 @@ func CreateTestCacheGroupParameters(t *testing.T) {
 	testData.CacheGroupParameterRequests = append(testData.CacheGroupParameterRequests, resp.Response...)
 }
 
+func CreateTestCacheGroupParametersAlreadyExist(t *testing.T) {
+
+	resp, _, _ := TOSession.GetAllCacheGroupParameters(client.RequestOptions{})
+	cachegroupParameters := resp.Response
+	getAllcgparameters := cachegroupParameters.CacheGroupParameters
+	if len(getAllcgparameters) < 1 {
+		t.Fatalf("No existing cachegroup parameters available to test duplicate functionality")
+	}
+	if getAllcgparameters[0].Parameter == nil && *getAllcgparameters[0].Parameter != 0 {
+		t.Fatalf("found nil/0 Value in Paramters")
+	}
+	if len(*getAllcgparameters[0].CacheGroup) < 1 {
+		t.Fatalf("found null Value in Cachegroup %d", len(*getAllcgparameters[0].CacheGroup))
+	}
+	parameterID := *getAllcgparameters[0].Parameter
+
+	//Get cachegroup id from cachegroup name
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("name", *getAllcgparameters[0].CacheGroup)
+
+	cgResponse, _, _ := TOSession.GetCacheGroups(opts)
+	if len(cgResponse.Response) < 1 {
+		t.Fatalf("No data available for GetCacheGroups by Name - Length %d", len(cgResponse.Response))
+	}
+	if cgResponse.Response[0].ID == nil && *cgResponse.Response[0].ID != 0 {
```

Comment:
The error says that it found a nil or 0 value for the Cache Group ID, but that will only get printed if the ID is both nil AND NOT 0. Which, by the way, also means that if the second half of this condition is ever checked, it will always segfault.

Reply:
Fixed it.